Source/WebCore/page/ContextMenuController.cpp

```diff
@@ -998,29 +1006,38 @@ void ContextMenuController::populate()
                 appendItem(StopItem, m_contextMenu.get());
                 appendItem(ReloadItem, m_contextMenu.get());
 #else
-                if (frame->page() && frame->page()->backForward().canGoBackOrForward(-1))
-                    appendItem(BackItem, m_contextMenu.get());
-
-                if (frame->page() && frame->page()->backForward().canGoBackOrForward(1))
-                    appendItem(ForwardItem, m_contextMenu.get());
-
-                // use isLoadingInAPISense rather than isLoading because Stop/Reload are
-                // intended to match WebKit's API, not WebCore's internal notion of loading status
-                if (loader.documentLoader()->isLoadingInAPISense())
-                    appendItem(StopItem, m_contextMenu.get());
-                else
-                    appendItem(ReloadItem, m_contextMenu.get());
+                if (frame->isMainFrame() || (frame->page() && !frame->ownerElement()->document().isPDFDocument())) {
```

Comment:
Did you mean to null check frame->ownerElement() instead of frame->page()?

Reply:
Added the null pointer check here. Also removed the frame->page() check since the more specific if branches are checking for them along with other checks below.